src/Gizmos/axisDragGizmo.ts

```diff
@@ -1,56 +1,30 @@
-import { Observer, Observable } from "../Misc/observable";
-import { Nullable } from "../types";
-import { PointerInfo } from "../Events/pointerEvents";
-import { Vector3, Matrix } from "../Maths/math.vector";
-import { TransformNode } from "../Meshes/transformNode";
-import { AbstractMesh } from "../Meshes/abstractMesh";
-import { Mesh } from "../Meshes/mesh";
-import { LinesMesh } from "../Meshes/linesMesh";
+import { Color3 } from "../Maths/math.color";
+import { Matrix, Vector3 } from "../Maths/math.vector";
 import { CylinderBuilder } from "../Meshes/Builders/cylinderBuilder";
-import { PointerDragBehavior } from "../Behaviors/Meshes/pointerDragBehavior";
-import { Gizmo } from "./gizmo";
+import { Mesh } from "../Meshes/mesh";
+import { TransformNode } from "../Meshes/transformNode";
 import { UtilityLayerRenderer } from "../Rendering/utilityLayerRenderer";
-import { StandardMaterial } from "../Materials/standardMaterial";
 import { Scene } from "../scene";
+import { Nullable } from "../types";
+import { DraggableGizmo } from "./draggableGizmo";
 import { PositionGizmo } from "./positionGizmo";
-import { Color3 } from '../Maths/math.color';
+import { StandardMaterial } from "../Materials/standardMaterial";
 /**
  * Single axis drag gizmo
  */
-export class AxisDragGizmo extends Gizmo {
-    /**
-     * Drag behavior responsible for the gizmos dragging interactions
-     */
-    public dragBehavior: PointerDragBehavior;
-    private _pointerObserver: Nullable<Observer<PointerInfo>> = null;
-    /**
-     * Drag distance in babylon units that the gizmo will snap to when dragged (Default: 0)
-     */
-    public snapDistance = 0;
-    /**
-     * Event that fires each time the gizmo snaps to a new location.
-     * * snapDistance is the the change in distance
-     */
-    public onSnapObservable = new Observable<{ snapDistance: number }>();
-
-    private _isEnabled: boolean = true;
-    private _parent: Nullable<PositionGizmo> = null;
-
+export class AxisDragGizmo extends DraggableGizmo {
     private _arrow: TransformNode;
-    private _coloredMaterial: StandardMaterial;
-    private _hoverMaterial: StandardMaterial;
 
     /** @hidden */
-    public static _CreateArrow(scene: Scene, material: StandardMaterial): TransformNode {
+    public static _CreateArrow(scene: Scene, material?: StandardMaterial): TransformNode {

```

Comment:
can be `Nullable<StandardMaterial> = null`. Then you don't need the `?? null` afterwards

Reply:
Fixed